GabbyTam_Fatima_functionpractice.js

```diff
@@ -0,0 +1,238 @@
+  //using built in exponent method 
+  //console.log(Math.pow(num,3));
+}
+console.log(calculateCube(2));
+
+
+// Part 1: 2 Write a function isAVowel that takes a character (i.e. a string of length 1) and returns true if it is a vowel, false otherwise. The vowel could be upper or lower case.
+// a, e, i, o, u
+function isAVowel(a) {
+  if (a === 'a' || a=== 'e' || a=== 'i'|| a=== 'o' || a=== 'u'){
+    return true;
+  } else {
+    return false;
+  }
+}
+
+//this works but its not efficient 
+console.log(isAVowel('e'));
+
+//Part 1: 3 Write a function getTwoLengths that accepts two parameters (strings). The function should return an array of numbers where each number is the length of the corresponding string.
+
+function getTwoLengths(first, second) {
+  let arrayNum= [];
+  arrayNum.push(first.length);
+  arrayNum.push(second.length);
+  return arrayNum;
+}
+//but why is in not .length-1?? 
+//answer: -1 and +1 comes into play when you are using for loops, declaring your variable at 0 or at 1 will make a difference as to what add on youll need to .length property 
+// .length counts string and arrays starting at index 0
+ 
+console.log(getTwoLengths("Hank", "Hippopopalous"));
+
+
+//Part 1: 4 Write a function getMultipleLengths that accepts a single parameter as an argument: an array of strings. The function should return an array of numbers where each number is the length of the corresponding string.
+
+const getMultipleLengths = (array) => {
+  let arrayOfNum= [];
+  for(let i=0; i<= array.length-1; i++){
+    arrayOfNum.push(array[i].length); 
+  }
+  return arrayOfNum;
+  //console.log(arrayOfNum);
+}
+
+console.log(getMultipleLengths(["hello", "what", "is", "up", "dude"]));
+
+//Part 2: 1 Define a function maxOfThree that takes three numbers as arguments and returns the largest of them. If all numbers are the same, it doesn't matter which one is returned. If the two largest numbers are the same, one of them should be returned.
+
+
+/*
+first attempt: 
+
+ const maxOfThree = (num1, num2, num3) => {
+  if(num1> num2 && num1> num3){
+    return num1;
+  }
+  if(num2> num1 && num2> num3){
+    return num2;
+  }
+  if(num3> num1 && num3 > num2){
+    return num3;
+  }
+  if(num1== num2== num3){
+    return num1;
+  }
+}
+
+console.log(maxOfThree(6, 9, 1));
+
+*/
+
+//attempt after googling 
+const maxOfThree = (num1, num2, num3) => {
+  let max= Math.max(num1,num2,num3);
+  return max;
+}
+
+console.log(maxOfThree(6, 9, 1));
+
+//Part 2: 2 Write a function printLongestWord that accepts a single argument, an array of strings. The method should return the longest word in the array. In case of a tie, the method should return the word that appears first in the array.
+
+/* This doesnt work because of the use of return, return stops/breaks the code
+
+
+function printLongestWord(arr){
+  for(let i=0; i<arr.length-1;i++){
+    if(arr[i].length> arr[i+1].length){
+      return arr[i];
+    } else {
+      return arr[i+1];
+    }
+  }
+}
```

Comment:
you're right that this doesn't work due to the `return` statements stopping the function. Instead of using `return`, consider declaring a variable inside the function and before the loop (between lines 113 and 114). Use this variable to hold the value of whatever word is longer, starting with the assumption that the word at `arr[0]` is the longest, and using the loop and conditional combo to check each following word and to replace the value of the variable as needed. Return the variable you declared before the loop after the loop is done